contracts/utils/AssociatedArrayLib.sol

```diff
@@ -10,44 +10,68 @@ library AssociatedArrayLib {
         uint256 _spacer;
     }
 
-    function _length(Array storage s, address account) private view returns (uint256 __length) {
+    function _slot(Array storage s, address account) private pure returns (bytes32 __slot) {
         assembly {
             mstore(0x00, account)
             mstore(0x20, s.slot)
-            __length := sload(keccak256(0x00, 0x40))
+            __slot := keccak256(0x00, 0x40)
+        }
+    }
+
+    function _length(Array storage s, address account) private view returns (uint256 __length) {
+        bytes32 slot = _slot(s, account);
+        assembly {
+            __length := sload(slot)
         }
     }
 
     function _get(Array storage s, address account, uint256 index) private view returns (bytes32 value) {
+        return _get(_slot(s, account), index);
+    }
+
+    function _get(bytes32 slot, uint256 index) private view returns (bytes32 value) {
         assembly {
-            mstore(0x00, account)
-            mstore(0x20, s.slot)
-            value := sload(add(keccak256(0x00, 0x40), mul(0x20, add(index, 1))))
+            //if (index >= _length(s, account)) revert AssociatedArray_OutOfBounds(index);
+            if iszero(lt(index, sload(slot))) {
+                mstore(0, 0x8277484f) // `AssociatedArray_OutOfBounds(uint256)`
+                mstore(4, index)
+                revert(0x1c, 0x24)
```

Comment:
@filmakarov two more locations could benefit from [the suggestion](https://github.com/erc7579/smartsessions/pull/66#discussion_r1747437948) to change `mstore(4, index)` -> `mstore(0x20, index)`. There is line 37 (within the `_get()` function) and also line 103 (within the `_remove()` function)

Reply:
yeah, thank you, fixed